src/components/Comments/Comment.js

```diff
@@ -111,6 +129,61 @@ const Comment = ({
     return COLORS.balance_gray;
   };
 
+  const username = comment?.data?.anon_user_info_color_name
+    ? `Anonymous ${comment.data?.anon_user_info_emoji_name}`
+    : user?.data?.username;
+
+  const onPressDM = async () => {
+    try {
+      setLoading({...loading, loadingDm: true});
+      if (!comment?.data?.anon_user_info_color_name) {
+        const channelName = [username, profile?.myProfile?.username].join(',');
+        const selectedUser = {
+          user: {
+            name: channelName,
+            image: comment?.user?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+          }
+        };
+        const members = [comment?.user?.id, profile.myProfile.user_id];
+        await createSignChat(members, selectedUser);
+      } else {
+        await sendMessageDM(comment?.id, 'comment', 'SIGNED');
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDm: false});
+    }
+  };
+
+  const onPressDMAnon = async () => {
+    try {
+      setLoading({...loading, loadingDmAnon: true});
+      await sendMessageDM(comment?.id, 'comment', 'ANONYMOUS');
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDmAnon: false});
+    }
+  };
+
+  const dataSheet = [
+    {
+      id: 1,
+      name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+      icon: <SendDMBlack />,
+      onPress: onPressDM
+    },
+    {
+      id: 2,
+      name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
+      icon: <SendDMAnonBlock />,
+      onPress: onPressDMAnon
+    }
+  ];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are a couple of issues here:

1. Error handling: In both `onPressDM` and `onPressDMAnon`, you're catching errors and logging them to the console, but there's no user-facing error handling. If an error occurs, the user won't be notified. Consider adding a mechanism to display error messages to the user.

2. Loading state: You're using `setLoading` to manage loading states for DMs, but you're spreading the previous state each time. This could potentially lead to unnecessary re-renders if other parts of the loading state change. Consider splitting these into separate state variables.

3. Direct mutation of refs: You're directly closing the bottom sheet by calling `refSheet.current.close()`. It's generally better to avoid direct mutations like this in React. Consider managing the visibility of the bottom sheet with a state variable instead.

```diff
-  const [loading, setLoading] = React.useState({
-    loadingDm: false,
-    loadingDmAnon: false
-  });
+  const [loadingDm, setLoadingDm] = React.useState(false);
+  const [loadingDmAnon, setLoadingDmAnon] = React.useState(false);

-  setLoading({...loading, loadingDm: true});
+  setLoadingDm(true);

-  refSheet.current.close();
+  setBottomSheetVisible(false);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->